SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestRepNotify/SpatialTestRepNotify.cpp

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialTestRepNotify.h"
+#include "SpatialGDK/Public/EngineClasses/SpatialNetDriver.h"
+
+#include "Net/UnrealNetwork.h"
+
+/**
+ * This test tests RepNotifies and shadow data implementation.
+ *
+ * The test includes 1 Server and 2 Client workers.
+ * The flow is as follows:
+ * - Setup:
+ *	- The test itself is used as the test Actor.
+ *  - It contains a number of replicated properties and OnRep functions that either set boolean flags confirming they have been called, or save the old value of the argument passed to the OnRep function.
+ * - Test:
+ *	- The Server changes the values of the replicated properties multiple times.
+ *  - The clients check whether the RepNotifies have or have not been called correctly.
+ * - Cleanup:
+ *	- None.
+ */
+
+ASpatialTestRepNotify::ASpatialTestRepNotify()
+	: Super()
+{
+	Author = "Miron";
```

Comment:
Is Andrei been Miron the whole time? Never seen them both on a video call together 🤔

Reply:
😂  Okay so the issue is this test is basically just polishing one old test that Miron originally wrote and had hidden in a strange dark PR somewhere. So I considered 'copyright' implications and decided that after all this is still Miron's test.

Reply:
So yeah I don't know what the 'correct' approach to this is.

Reply:
Hah - squash your names together - "Miron+Andrei"